### PR TITLE
Create typings for @dsinjs/binary-tree

### PR DESCRIPTION
**Pull request by siddhesh321995**
Please fill in this template.

- @dsinjs/binary-tree
- tested using npm test @dsinjs/binary-tree

Select one of these and delete the others:

If adding a new definition:
- [ X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.

### CODE REVIEW

File: types/dsinjs__binary-tree/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "lib": [
+      "ES2018",
+      "ES2019",
+      "es2017",
+      "es7",
+      "es6",
+      "es5",
+      "dom"
+    ],
+    "strict": true,
```
Comment (amcasey) [inline types/dsinjs__binary-tree/tsconfig.json:14]: I think we've been trying to converge on this convention
```json
        "paths":{
            "@dsinjs/*": ["dsinjs__*"]
        },
```

File: types/dsinjs__binary-tree/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "module": "commonjs",
+    "lib": [
+      "ES2018",
```
Comment (amcasey) [inline types/dsinjs__binary-tree/tsconfig.json:6]: Why were all of these needed?

File: types/dsinjs__binary-tree/tslint.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (amcasey) [inline types/dsinjs__binary-tree/tslint.json:3]: Imposing additional linting constraints on an individual package tends to make it harder for others to contribute.  Are these important?

File: types/dsinjs__binary-tree/tslint.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        "max-line-length": {
+            "options": [180]
+        },
+        "new-parens": true,
+        "no-conditional-assignment": true,
+        "no-consecutive-blank-lines": false,
```
Comment (amcasey) [inline types/dsinjs__binary-tree/tslint.json:9]: Why suppress lint rules?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/dsinjs__binary-tree/tsconfig.json (hunk 1 above)
Comment (siddhesh321995) [reply types/dsinjs__binary-tree/tsconfig.json:14]: Sure I will follow these rules,
Sorry was not aware of this.

File: types/dsinjs__binary-tree/tsconfig.json (hunk 2 above)
Comment (siddhesh321995) [reply types/dsinjs__binary-tree/tsconfig.json:6]: I guess basic lib like dom, ES2017 will also work, will update these changes.

File: types/dsinjs__binary-tree/tslint.json (hunk 3 above)
Comment (siddhesh321995) [reply types/dsinjs__binary-tree/tslint.json:3]: Which of these addition constraints do you recommend removing?

File: types/dsinjs__binary-tree/tslint.json (hunk 4 above)
Comment (siddhesh321995) [reply types/dsinjs__binary-tree/tslint.json:9]: I will use the default constraint values for these  3 rules

File: types/dsinjs__binary-tree/tslint.json (hunk 3 above)
Comment (amcasey) [reply types/dsinjs__binary-tree/tslint.json:3]: Generally speaking, the contents of this file should be exactly `{ "extends": "dtslint/dt.json" }`.

File: types/dsinjs__binary-tree/tsconfig.json (hunk 1 above)
Comment (amcasey) [reply types/dsinjs__binary-tree/tsconfig.json:10]: Not your fault - it's new and undocumented.  Thanks for updating!

**Review by amcasey — APPROVED** [review]
Ideally, the lint rules would be further cleaned up, but this looks basically correct.

File: types/dsinjs__binary-tree/tslint.json (hunk 3 above)
Comment (siddhesh321995) [reply types/dsinjs__binary-tree/tslint.json:3]: I see, thank you for your kind reply

**Comment by siddhesh321995** [comment]
Ready to merge

**Pull request merged**